packages/iTwinUI-react/src/core/utils/components/Resizer.tsx

```diff
@@ -0,0 +1,298 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { getBoundedValue, getTranslateValues, getWindow } from '../functions';
+
+export type ResizerProps = {
+  /**
+   * Ref of the element that is being resized.
+   */
+  elementRef: React.RefObject<HTMLElement>;
+  /**
+   * Ref of the container element in order to avoid resizing past container boundaries.
+   * If not passed, viewport will be used.
+   */
+  containerRef?: React.RefObject<HTMLElement>;
+  /**
+   * Callback that is being called on resize end.
+   * Useful to preserve state if element is being closed.
+   */
+  setStyle?: (style: React.CSSProperties) => void;
```

Comment:
would it make more sense to call this `onResize`? in case we want to pass more things when resize ends

Reply:
I would expect `onResize` to be called on every `onMouseMove` call, maybe then it could be `onResizeEnd`

Reply:
`onResizeEnd` works 👍

Reply:
Renamed